shop/src/main/java/ro/msg/learning/shop/controller/exception/ControllerExceptionHandler.java

```diff
@@ -7,11 +7,17 @@
 import org.springframework.web.context.request.WebRequest;
 import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
 import ro.msg.learning.shop.exception.EntityNotFoundException;
+import ro.msg.learning.shop.exception.NoSuitableDeliveryStocksFoundException;
 
 @RestControllerAdvice
 public class ControllerExceptionHandler extends ResponseEntityExceptionHandler {
     @ExceptionHandler(EntityNotFoundException.class)
     public ResponseEntity<Object> handleEntityNotFound(RuntimeException exception, WebRequest request) {
         return new ResponseEntity<>(new ErrorResponse(exception.getMessage()), HttpStatus.NOT_FOUND);
     }
+
+    @ExceptionHandler(NoSuitableDeliveryStocksFoundException.class)
+    public ResponseEntity<Object> handleProductOrderError(RuntimeException exception, WebRequest request) {

```

Comment:
Not a big fan of ResponseEntity<Object>. However, i see that it is fitting here, but I bet you can improve this in the future to make it as specific as possible :).
⭐ gold star for using exception handler and spring further spring annotations on your own.